central/rbac/k8srolebinding/datastore/datastore_impl.go

```diff
@@ -26,16 +26,30 @@ func (d *datastoreImpl) GetRoleBinding(ctx context.Context, id string) (*storage
 }
 
 func (d *datastoreImpl) SearchRoleBindings(ctx context.Context, q *v1.Query) ([]*v1.SearchResult, error) {
+	if q == nil {
+		q = searchPkg.EmptyQuery()
+	}
+	q = q.CloneVT()
```

Comment:
Be consistent.  Probably better to create a clone and work on the clone like you did above in roles.

qClone := q.CloneVT()

Also when you finish, these you need to go back through and make sure they all use the clone.